### PR TITLE
Simplify and improve path lookup code and add `PORTABLE` CMake option as per my suggestions in #209

### PR DESCRIPTION
**Pull request by ntninja**
Simplify and improve path lookup code and add `PORTABLE` CMake option as per my suggestions in #209

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,6 +5,8 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules/")
 set(PLATFORM "pc" CACHE STRING "Platform to build for")
 set_property(CACHE PLATFORM PROPERTY STRINGS pc vita switch)
 
+set(PORTABLE OFF CACHE BOOL "Build location-independent binary on Linux/Unix?")
```
Comment (orbea) [inline CMakeLists.txt:8]: Why not `option`?

https://cmake.org/cmake/help/latest/command/option.html

**Comment by isage** [comment]
That breaks windows build, no?
Windows version isn't built with cmake, but rather with  VS project (because cmake-generated projects are ugh), so DATADIR should be set in VS project

File: CMakeLists.txt (hunk 1 above)
Comment (ntninja) [reply CMakeLists.txt]: Because I didn't know about that one. Doc implies it does the same thing as this usage of `set`, but sure looks better. Fixed!

**Comment by ntninja** [comment]
> That breaks windows build, no?
> Windows version isn't built with cmake, but rather with VS project (because cmake-generated projects are ugh), so DATADIR should be set in VS project

Oh that's what duplicating those paths down there achieved!

By the looks of it, it appears to me that Windows and Apple code paths are exactly the same, no? That is: Resources are all relative to the main binary installation path. If that is the case just tell me which define check I should add for Windows and I'll fix up the code accordingly. Using `DATADIR` for Windows isn't add good idea since binaries need to be location-independent there as well (both for ZIP and possible future installer).

Also, it looks like there was no support on Windows for user translations/mods? If so, merging the previous Windows code with macOS should automatically enable support for that in `%APPDATA%\nxengine\nxengine-evo\data`. If there was such a thing before please tell me how and where it was so that I can include that too.

**Comment by ntninja** [comment]
On a more general note it appears to me that using relative paths (with `../share/nxengine` prefix on Linux/BSD) is the much saner default and hard-coding paths is only really interesting for some Linux/BSD distros. As such, I'd suggest making `PORTABLE` the default and instead adding an inverse flag for this particular user group to use. Any objections?

**Comment by ntninja** [comment]
Finally, @isage: Am I correctly assuming all this time that the `__linux__` macro is defined iff we are in a `UNIX AND NOT APPLE AND NOT ANDROID` environment. That is, this macro will be defined on all non-macOS Unix-like systems? If that is NOT the case, shouldn't we replace this check with a CMake generated define based on the above CMake rule instead?

**Comment by isage** [comment]
`__linux__` is defined on all linux-derived systems. E.g. android.
See https://blog.kowalczyk.info/article/j/guide-to-predefined-macros-in-c-compilers-gcc-clang-msvc-etc..html

Windows was supported by those https://github.com/nxengine/nxengine-evo/blob/b8b58c7f0bae52b5d911f780d0197d073c303776/src/ResourceManager.cpp#L154-L160
as windows port is always portable.That also worked for OSX, because i'm a moron and because of chdir in main. 

Instead of relative paths we should probably use https://wiki.libsdl.org/SDL_GetBasePath
And i propose we leave that relative fallback in. That'll work for win/osx/vita and for portable linux/osx/etc. builds.

So, in the end i propose following order:
1) Search relative to SDL_GetPrefPath on all platforms except SWITCH (this allows user content in ~/.local etc.)
2) Search in DATADIR if defined. (that's for packaging)
3) Search relative to SDL_GetBasePath on all platforms, except SWITCH (that always allows portable builds)
(SWITCH is exception, because there's no  there, but we can probably simplify code by emulating those two: e.g. SDL_GetPrefPath would be empty, and SDL_GetBasePath will be `romfs:`)

That'll also allow to remove chdir from main and vita-specific codepath from ResourceManager::getPrefPath (and switch too if we redefine SDL_GetPrefPath/SDL_GetBasePath and provide own functions instead)

By doing that we'll drop shitton of platform ifdefs and will just use same code for all platforms.

**Comment by ntninja** [comment]
Thanks for the detailed explanation, I'm working on it!

However: Your explanation sounds like I should use the desktop codepaths on Vita. Is that correct? They all start with `ux0:` and `app0:` right now.

**Comment by isage** [comment]
Yeah. SDL_GetBaseDir/PrefDir will take care of that. Older vita sdl port
(e.g. when nxevo port was done) didn't support that, now it does.

**Comment by isage** [comment]
I've set up discord server, btw, as an easier way to discuss things. Link is in project about.

**Comment by ntninja** [comment]
@isage: Thanks, but no thanks. I'm not going to join discord like ever given their “data privacy standards”. May I interest you in Matrix/Element instead?

**Comment by isage** [comment]
Well, that's sad but understandable. Matrix or anything else is sadly one
too much chat software for me. IRC, maybe?

**Comment by orbea** [comment]
> I've set up discord server, btw, as an easier way to discuss things.

https://sneak.berlin/20200220/discord-is-not-an-acceptable-choice-for-free-software-projects/
https://stallman.org/discord.html

**Comment by ntninja** [comment]
IRC is fine, Matrix has a great IRC bridge. :slightly_smiling_face: 

@orbea: No need to push here. They meant well and what they said wasn't wrong. Didn't even push back when I said I wouldn't do it.

**Comment by isage** [comment]
We can continue off-topic there https://github.com/nxengine/nxengine-evo/issues/215

**Comment by ntninja** [comment]
It passes again! :grinning:

**Comment by isage** [comment]
Merged manually with additions.

**Pull request closed without merging**